libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranularPermissions/GranularGranteeGroupItem.tsx

```diff
@@ -1,31 +1,51 @@
 // (C) 2022-2023 GoodData Corporation
 
-import React, { useMemo } from "react";
+import React, { useCallback, useMemo } from "react";
 import { useIntl } from "react-intl";
 import cx from "classnames";
 
 import { GranteeGroupIcon } from "../GranteeIcons";
-import { GranteeItem, IGranularGranteeGroup } from "../types";
+import { DialogModeType, GranteeItem, IGranularGranteeGroup } from "../types";
 import { getGranteeItemTestId, getGranteeLabel } from "../utils";
 import { CurrentUserPermissions } from "../../types";
 
 import { GranularPermissionsDropdownWithBubble } from "./GranularPermissionsDropdown";
 import { usePermissionsDropdownState } from "./usePermissionsDropdownState";
 import { getGranteePossibilities } from "./permissionsLogic";
+import { useComponentInteractionContext } from "../ComponentInteractionContext";
 
 interface IGranularGranteeGroupItemProps {
     grantee: IGranularGranteeGroup;
     currentUserPermissions: CurrentUserPermissions;
     isDashboardLocked: boolean;
+    mode: DialogModeType;
     onChange: (grantee: GranteeItem) => void;
     onDelete: (grantee: GranteeItem) => void;
 }
 
 export const GranularGranteeGroupItem: React.FC<IGranularGranteeGroupItemProps> = (props) => {
-    const { grantee, currentUserPermissions, isDashboardLocked, onChange, onDelete } = props;
+    const { grantee, currentUserPermissions, isDashboardLocked, onChange, onDelete, mode } = props;
+    const { permissions } = grantee;
     const { isDropdownOpen, toggleDropdown } = usePermissionsDropdownState();
+    const onInteraction = useComponentInteractionContext();
     const intl = useIntl();
 
+    const handleToggleDropdown = useCallback(() => {
```

Comment:
I would create custom hook outside of this component (in the same file) and hide the creation of callback there. It polutes the component IMHO. You can use component props as its parameter.
Something like `const handleToogleDropdown = useToggleDropdownWithOnInteraction(props);`. Possibly you will find better name for it than this horridness.


Same goes for `libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranularPermissions/GranularGranteeUserItem.tsx`

Reply:
I separated all the logic into one hook with multiple callbacks, thanks